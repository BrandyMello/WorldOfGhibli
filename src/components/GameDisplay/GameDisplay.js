import React from 'react';
import Game from '../../containers/Game/Game';
import PropTypes from 'prop-types';

const GameDisplay = ({characters, player}) => {
  console.log({player})
  const characterNames = characters.map(character => character.name);

  const mappedGames = characters.map((character, index) => {
    return <Game
      index={index}
      key={character.id}
      name={character.name}
      gender={character.gender}
      age={character.age}
      eyeColor={character.eye_color}
      hairColor={character.hair_color}
      characterNames={characterNames}
      player={player}
    />
  })

  return (
    <div className="background">
      <h2>Unscramble the character name for a win, {player}</h2>
      {mappedGames}
    </div>
  )
}

export default GameDisplay

GameDisplay.propTypes = {
  characters: PropTypes.array.isRequired,
  player: PropTypes.array
}