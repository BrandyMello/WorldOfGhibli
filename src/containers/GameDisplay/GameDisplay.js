import React from 'react';
import Game from '../Game/Game';
import images from '../../images/images.js';

const GameDisplay = ({characters}) => {
  const characterNames = characters.map(character => character.name);
  const mappedGames = characters.map((character) => {
    return <Game
      key={character.id}
      name={character.name}
      gender={character.gender}
      age={character.age}
      eyeColor={character.eye_color}
      hairColor={character.hair_color}
      characterNames={characterNames}
    />
  })

  return (
    <div className="background">
      {mappedGames}
    </div>
  )
}

export default GameDisplay
