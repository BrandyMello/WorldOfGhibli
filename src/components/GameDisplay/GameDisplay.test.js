import React from 'react';
import GameDisplay from './GameDisplay';
import { shallow } from 'enzyme';

describe('GameDisplay', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <GameDisplay
        characters={
          [
            {
              id: 1,
              name: 'Eboshi'
            },
            {
              id: 2,
              name: 'Ashikata'
            }
          ]
        }
        player='Brandy'
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
