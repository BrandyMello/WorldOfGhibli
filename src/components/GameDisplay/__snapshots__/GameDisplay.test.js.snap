// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameDisplay should match the snapshot 1`] = `
<div
  className="background"
>
  <h2>
    Unscramble the character name for a win, 
    Brandy
  </h2>
  <Connect(Game)
    characterNames={
      Array [
        "Eboshi",
        "Ashikata",
      ]
    }
    index={0}
    key="1"
    name="Eboshi"
    player="Brandy"
  />
  <Connect(Game)
    characterNames={
      Array [
        "Eboshi",
        "Ashikata",
      ]
    }
    index={1}
    key="2"
    name="Ashikata"
    player="Brandy"
  />
</div>
`;
